includes/wc-product-functions.php

```diff
@@ -751,3 +751,123 @@ function wc_get_product_visibility_options() {
 		'hidden'  => __( 'Hidden', 'woocommerce' ),
 	) );
 }
+
+/**
+ * Get relateed products.
+ * @param  integer  $product_id
+ * @param  integer  $limit
+ * @param  array    $exclude_ids
+ * @return array
+ */
+function wc_get_related_products( $product_id, $limit = 5, $exclude_ids = array() ) {
+	global $wpdb;
+
+	$product_id     = absint( $product_id );
+	$exclude_ids    = array_map( 'absint', array_merge( array( 0, $product_id ), $exclude_ids ) );
```

Comment:
I think the `absint` sanitization should be inside `wc_get_related_products_query` (or in both places). Now that `wc_get_related_products_query` is a public global function, instead of a private function, people might use it directly to generate a query. So`$exclude_ids`, `$cats_array`, and `$tags_array` should be escaped as late as possible I think to prevent someone from making a mistake.